providers/docker/src/airflow/providers/docker/operators/docker_swarm.py

```diff
@@ -218,9 +218,8 @@ def _run_service(self) -> None:
         if self.service and self._service_status() == "complete":
             self.tasks = self.cli.tasks(filters={"service": self.service["ID"]})
             for task in self.tasks:
-                container_id = task["Status"]["ContainerStatus"]["ContainerID"]
-                container = self.cli.inspect_container(container_id)
-                self.containers.append(container)
+                docker_service = self.cli.inspect_service(task["ServiceID"])
+                self.containers.append(docker_service)
```

Comment:
Would it be better to check the key for better compatibility?

```suggestion
    if "ServiceID" in task:
        docker_service = self.cli.inspect_service(task["ServiceID"])
        self.containers.append(docker_service)
    elif "Status" in task and "ContainerStatus" in task["Status"]:
        container_id = task["Status"]["ContainerStatus"]["ContainerID"]
        container = self.cli.inspect_container(container_id)
        self.containers.append(container)
```

Reply:
Hello, thank you for picking up the MR.

What kind of combatibility are we trying to achieve? Unless I am missing something, I would say that `inspect_container()` is _wrong_ in general because it works only in 1-node deployments. I don't see why/when `ServiceID` is not available, since we are working with services specifically, which is the fundamental resource of Docker Swarm.

Reply:
> What kind of combatibility are we trying to achieve?

I'm not sure will the change introduce breaking change for Docker Swarm side (e.g. not compatible for lower Docker Swarm Version )

>  I would say that inspect_container() is wrong in general because it works only in 1-node deployments. I don't see why/when ServiceID is not available, since we are working with services specifically, which is the fundamental resource of Docker Swarm.

Thanks for the clarification! If this is the case, then the change LGTM.
Additionally, it would be nice to test this change with real Airflow setup, then this PR is ready to go. Thanks!

Reply:
Actually this popped up at my work. We are using Airflow to spawn containers in a 2-node Swarm Cluster (the plan is to add more nodes eventually).
Currently until this is merged, I am shadowing docker_swarm.py using volume in docker compose.
So it's tested in a real world system, specifically with docker version 28.4.0.

Reply:
Nice! The final thing is to fix the static check for formatting, thanks!

```bash
uv tool install prek
prek ruff --all-files && prek run ruff-format --all-files
```

Reply:
Fixed! Thanks @jason810496.